### DISCUSSION TITLE
Can I save a 2-bit or 3-bit BMP?

### DISCUSSION THREAD

**Discussion by davyike** [SixLabors/ImageSharp#2218] [Q&A]
I'm building a webserver for an [e-ink display](https://www.crowdsupply.com/soldered/inkplate-10) that supports 2 or 3-bit grayscale BMP images. Is there any way I can output images like this?

I'm saving 4-bit images as below but the micropython display code can't handle them well.
```
img.SaveAsBmp(ms, new BmpEncoder()
{
    SupportTransparency = false,
    BitsPerPixel = BmpBitsPerPixel.Pixel4,
});
```

**Comment by brianpopow** [comment]
We dont support encoding 2- or 3-Bit bitmaps, because in the specifications those bit depth are not mentioned.

For example:
- [Microsoft Windows Bitmap File Format](https://www.fileformat.info/format/bmp/egff.htm)
- [OS/2 Bitmap File Format](https://www.fileformat.info/format/os2bmp/egff.htm)
- [Microsoft, Types of Bitmaps](https://docs.microsoft.com/en-us/windows/win32/gdiplus/-gdiplus-types-of-bitmaps-about), note in the intro to bitmap types 2 bit palette is mentioned, but in the Graphics File Formats section it's not.
- Comprehensive List of all [Bitmap types](http://entropymine.com/jason/bmpsuite/bmpsuite/html/bmpsuite.html): Here 2 bit per pixels is mentioned as "questional" and that it was legal on Windows CE.

I dont think we should add support for 3 bit bitmaps, since I cant find mentions for those anywhere, but there can be made an argument for adding 2 bit per pixel bitmaps. @JimBobSquarePants: what is your opinion on adding support for 2 bit per pixel bitmaps?

**Reply by davyike** [reply]
I managed to find a solution to my problem - by using the Arduino library rather than micropython, they have functions that can display a PNG. So I'm now just rendering a PNG from ImageSharp. That said - if this is a small change, it could be handy to others in the future.

Thank you for your response.

**Reply by JimBobSquarePants** [reply]
@brianpopow The first link you have from the MS docs explicitly lists the bit depths which excludes 2 bit however since it appears from the last link that browsers can display bitmaps of that depth so there must be support out there. If it's not difficult I would say we add it.

**Reply by JimBobSquarePants** [reply]
Now possible via https://github.com/SixLabors/ImageSharp/pull/2220 Thanks @brianpopow
